front/pages/api/w/[wId]/invitations/index.ts

```diff
@@ -154,41 +156,63 @@ async function handler(
         });
       }
       const existingMembers = await getMembers(auth);
-      const unconsumedInvitations = await getRecentPendingOrRevokedInvitations(
+      const unconsumedInvitations = await getRecentPendingAndRevokedInvitations(
         auth
       );
       if (
-        unconsumedInvitations.length >=
+        unconsumedInvitations.pending.length >=
         MAX_UNCONSUMED_INVITATIONS_PER_WORKSPACE_PER_DAY
       ) {
         return apiError(req, res, {
           status_code: 400,
           api_error: {
             type: "invalid_request_error",
-            message: `Too many unconsumed invitations. Please ask your members to consume their invitations before sending more.`,
+            message: `Too many pending invitations. Please ask your members to consume their invitations before sending more.`,
           },
         });
       }
-      const emailsWithRecentUnconsumedInvitations = new Set(
-        unconsumedInvitations.map((i) => i.inviteEmail.toLowerCase().trim())
-      );
-      if (
-        invitationRequests.some((r) =>
-          emailsWithRecentUnconsumedInvitations.has(
+
+      const emailsWithRecentUnconsumedInvitations = new Set([
+        ...unconsumedInvitations.pending.map((i) =>
+          i.inviteEmail.toLowerCase().trim()
+        ),
+        ...unconsumedInvitations.revoked.map((i) =>
+          i.inviteEmail.toLowerCase().trim()
+        ),
+      ]);
+      const emailsToSendInvitations = invitationRequests.filter(
+        (r) =>
+          !emailsWithRecentUnconsumedInvitations.has(
             r.email.toLowerCase().trim()
           )
-        )
+      );
+      if (
+        !emailsToSendInvitations.length &&
+        !unconsumedInvitations.revoked.length &&
+        invitationRequests.length > 0
       ) {
         return apiError(req, res, {
           status_code: 400,
           api_error: {
-            type: "invalid_request_error",
-            message: `Some of the emails have already received an invitation in the last 24 hours. Please wait before sending another invitation.`,
+            type: "invitation_already_sent_recently",
+            message: `These emails have already received an invitation in the last 24 hours. Please wait before sending another invitation.`,
           },
         });
       }
+      await MembershipInvitation.update(
```

Comment:
This should not live here. Please move to `front/lib/api`.

Reply:
Moved